app/controllers/sessions_controller.rb

```diff
@@ -1,8 +1,26 @@
 class SessionsController < ApplicationController
   def create
-    @user = request.env['omniauth.auth']['extra']['raw_info']
-    user = User.find_by usr_auth_sch_id: @user['internal_id']
+    oauth_data = request.env['omniauth.auth']['extra']['raw_info']
+    user = User.find_by usr_auth_sch_id: oauth_data['internal_id']
+    if !user
+      session[:oauth_data] = oauth_data
+      redirect_to '/register'
+    else
+      session[:user] = user.id
+      redirect_to '/'
+    end
+  end
+
+  def register
+    user = User.create(usr_auth_sch_id: session[:oauth_data]["internal_id"],
```

Comment:
`User.create` csendben elhal hogy ha validacios hiba van es a felhasznalo nem ertesul rola. Ha itt elmeletileg sosem lehet validacio hiba akkor hasznalhatok a felkialotjeles verziot: `User.create!(...)`.

Valamint a `session[:oauth_data]`-t lehet erdemes elmenti egy valtozoba, mert ez igy eleg furcsan mutat.
Tovabba `request.params` helyett csak siman `params` is eleg es ez a Rails-es konvecio is.